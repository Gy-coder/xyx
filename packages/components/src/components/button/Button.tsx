import { FC, ButtonHTMLAttributes } from "react";

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {}

const Button: FC<ButtonProps> = (props) => {
  const { children, ...rest } = props;
  return <button {...rest}>{children}</button>;
};

export default Button;
