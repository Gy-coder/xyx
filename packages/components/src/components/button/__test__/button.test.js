import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Button from "../Button";

describe("test button component", () => {
  it("button should render correctly", () => {
    const { container, asFragment } = render(<Button>+1</Button>);
    const el = container.getElementsByClassName("g-button")[0];
    expect(el).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });
  it("can be click", () => {
    const jestFn = jest.fn();
    const { container } = render(<Button onClick={jestFn}>+1</Button>);
    const el = container.getElementsByClassName("g-button")[0];
    expect(jestFn).not.toBeCalled();
    fireEvent.click(el);
    expect(jestFn).toBeCalledTimes(1);
  });
});
