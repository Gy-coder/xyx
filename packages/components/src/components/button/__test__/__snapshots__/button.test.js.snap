// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button component button should render correctly 1`] = `
<DocumentFragment>
  <button
    class="g-button g-button-middle g-button-default"
    type="button"
  >
    <span
      class="g-button-content"
    >
      +1
    </span>
  </button>
</DocumentFragment>
`;
